posthog/hogql/test/test_expr_parser.py

```diff
@@ -0,0 +1,96 @@
+from posthog.hogql.expr_parser import ExprParserContext, translate_hql
+from posthog.test.base import APIBaseTest, ClickhouseTestMixin  # , test_with_materialized_columns
+
+
+class TestExprParser(ClickhouseTestMixin, APIBaseTest):
+    def test_hogql_literals(self):
+        self.assertEqual(translate_hql("1 + 2"), "plus(1, 2)")
+        self.assertEqual(translate_hql("-1 + 2"), "plus(-1, 2)")
+        self.assertEqual(translate_hql("-1 - 2 / (3 + 4)"), "minus(-1, divide(2, plus(3, 4)))")
+        self.assertEqual(translate_hql("1.0 * 2.66"), "multiply(1.0, 2.66)")
+        self.assertEqual(translate_hql("'string'"), "'string'")
+        self.assertEqual(translate_hql('"string"'), "'string'")
+
+    # @test_with_materialized_columns(["$browser"])
+    def test_hogql_fields_and_properties(self):
+        self.assertEqual(
+            translate_hql("properties.bla"),
+            "replaceRegexpAll(JSONExtractRaw(properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(
+            translate_hql("properties['bla']"),
+            "replaceRegexpAll(JSONExtractRaw(properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(
+            translate_hql('properties["bla"]'),
+            "replaceRegexpAll(JSONExtractRaw(properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(
+            translate_hql("properties.$bla"),
+            "replaceRegexpAll(JSONExtractRaw(properties, '$bla'), '^\"|\"$', '')",
+        )
+        # self.assertEqual(translate_hql("properties['$browser']"), "\"mat_$browser\"")
+        self.assertEqual(
+            translate_hql("person.properties.bla"),
+            "replaceRegexpAll(JSONExtractRaw(person_properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(translate_hql("uuid"), "uuid")
+        self.assertEqual(translate_hql("event"), "event")
+        self.assertEqual(translate_hql("timestamp"), "timestamp")
+        self.assertEqual(translate_hql("distinct_id"), "distinct_id")
+        self.assertEqual(translate_hql("person_id"), "person_id")
+        self.assertEqual(translate_hql("person.created_at"), "person_created_at")
+
+    def test_hogql_methods(self):
+        self.assertEqual(translate_hql("total()"), "count(*)")
+
+    def test_hogql_expr_parse_errors(self):
```

Comment:
Can we also have some tests against SQL injection with `;`?

Reply:
added one that tests that 
```py
        self._assert_value_error("1;2", "Module body must contain only one 'Expr'")
```